pkg/db/db.go

```diff
@@ -0,0 +1,93 @@
+package db
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+)
+
+type Database struct {
+	tableName string
+	pool      *pgxpool.Pool
+}
+
+func NewClient(ctx context.Context, connString, providerTableName string) (*Database, error) {
+	pool, err := pgxpool.New(ctx, connString)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := pool.Ping(ctx); err != nil {
+		return nil, err
+	}
+
+	return &Database{
+		pool:      pool,
+		tableName: providerTableName,
+	}, nil
+}
+
+func (pg *Database) Close() {
+	pg.pool.Close()
+}
+
+func (pg *Database) ListenForNotifications(ctx context.Context, connString, providerTableName string, fn func(payload string)) error {
```

Comment:
Why are we trying to implement a generic listen? I would prefer to be specific to the event/notification the we are trying to listen to.

Also, no need to provide the connection string or the providersTableName as you already have access to the Database object

```suggestion
func (pg *Database) OnProvider(ctx context.Context, fn func(providerName string)) error {
```